packages/components/src/molecules/SearchProducts/SearchProductItemControl.tsx

```diff
@@ -0,0 +1,127 @@
+import React, {  forwardRef, HTMLAttributes, useCallback, useState } from 'react'
+import { Badge, Icon, IconButton, Input, Loader, QuantitySelector } from '../..'
+
+import type { ReactNode, MouseEvent } from 'react'
+
+type StatusButtonAddToCartType = 'default' | 'inProgress' | 'completed'
+
+export interface SearchProductItemControlProps
+  extends Omit<HTMLAttributes<HTMLDivElement>, 'children' | 'onClick'> {
+  children: ReactNode
+  availability: boolean
+  hasVariants: boolean
+  skuMatrixControl: ReactNode
+  quantity: number
+  onClick?(e: MouseEvent<HTMLButtonElement>): void
+  onChangeQuantity(value: number): void
+}
+
+const SearchProductItemControl = forwardRef<
+  HTMLDivElement,
+  SearchProductItemControlProps
+>(function SearchProductItemControl(
+  {
+    availability,
+    children,
+    hasVariants,
+    skuMatrixControl,
+    quantity,
+    onClick,
+    onChangeQuantity,
+    ...otherProps
+  },
+  ref
+) {
+  const [statusAddToCart, setStatusAddToCart] =
+    useState<StatusButtonAddToCartType>('default')
+  function stopPropagationClick(e: MouseEvent) {
+    e.preventDefault()
+    e.stopPropagation()
+  }
+  function handleAddToCart(event: MouseEvent<HTMLButtonElement>) {
+    if (onClick) {
+      setStatusAddToCart('inProgress')
+
+      setTimeout(() => {
+        setStatusAddToCart('completed')
+        onClick(event)
+      }, 1000)
+
+      setTimeout(() => {
+	setStatusAddToCart('default')
+	onChangeQuantity(1)
+      }, 2000)
+    }
+  }
+	
+  const getIcon = useCallback(() => {
+    switch (statusAddToCart) {
+      case 'inProgress':
+        return <Loader />
+      case 'completed':
+        return <Icon name="Checked" width={24} height={24} />
+      default:
+        return <Icon name="ShoppingCart" width={24} height={24} />
+    }
+  }, [statusAddToCart])
+
+	const showSKUMatrixControl = availability && hasVariants;
+	const isMobile = window.innerWidth <= 768
```

Comment:
todo: Can we move these 2 const declaration to the top? (below `statusAddToCart` const)